plugins/modules/foreman_scc_account.py

```diff
@@ -0,0 +1,177 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: foreman_scc_account
+short_description: Manage Foreman SccAccount
+description:
+  - "Manage Foreman SccAccount Entities"
+  - "This module requires the foreman_scc_manager plugin set up in the server"
+  - "See: U(https://github.com/ATIX-AG/foreman_scc_manager)"
+author:
+  - "Manisha Singhal (@manisha15) ATIX AG"
+options:
+  name:
+    description: Name of the scc_account
+    required: true
+    type: str
+  login:
+    description: Login id of scc_account
+    required: false
+    type: str
+  scc_account_password:
+    description: Password of scc_account
+    required: false
+    type: str
+  base_url:
+    description: URL of SUSE for scc_account
+    required: false
+    type: str
+  interval:
+    description: Interval for syncing scc_account
+    required: false
+    type: str
+    choices: ["never", "daily", "weekly", "monthly"]
+  sync_date:
+    description: Last Sync time of scc_account
+    required: false
+    type: str
+  organization:
+    description: Name of related organization
+    type: str
+    required: true
+  test_connection:
+    description: Test scc_account credentials that connects to the server
+    required: false
+    default: false
+    type: bool
+  updated_name:
+    description: Name to be updated of scc_account
+    type: str
+  state:
+    description: State of the scc_account
+    default: present
+    choices: ["present", "absent", "synced"]
+    type: str
+extends_documentation_fragment:
+  - foreman
+  - foreman.organization
+'''
+
+EXAMPLES = '''
+- name: "Create an scc_account"
+  foreman_scc_account:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Test"
+    login: "abcde"
+    scc_account_password: "12345"
+    base_url: "https://scc.suse.com"
+    state: present
+
+- name: "Update a scc_account"
+  foreman_scc_account:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Test1"
+    state: present
+
+- name: "Delete a scc_account"
+  foreman_scc_account:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "Test"
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+
+from ansible.module_utils.foreman_helper import KatelloEntityAnsibleModule
+
+
+def main():
+    module = KatelloEntityAnsibleModule(
+        entity_spec=dict(
+            name=dict(required=True),
+            updated_name=dict(),
+            login=dict(),
+            scc_account_password=dict(no_log=True, flat_name='password'),
+            base_url=dict(),
```

Comment:
should this default to `https://scc.suse.com`?

Reply:
That's the default on the server side, and providing a default here would reset that value if unspecified.

Reply:
ah, good!